### PR TITLE
Defloating blas

### PR DESCRIPTION
**Pull request by evsmithx**
Removed float constructors from blas.

### CODE REVIEW

File: libs/math/src/math/linalg/blas/gemm_tn_novector.cpp [hunk 1]
```diff
@@ -38,21 +38,21 @@ void Blas<S, Signature(_C <= _alpha, _A, _B, _beta, _C),
   std::size_t i;
   std::size_t j;
   if ((c.height() == 0) ||
-      ((c.width() == 0) || (((alpha == static_cast<Type>(0.0)) || (a.height() == 0)) &&
-                            (beta == static_cast<Type>(1.0)))))
+      ((c.width() == 0) || (((alpha == fetch::math::Type<Type>("0.0")) || (a.height() == 0)) &&
```
Comment (vhavrylov-ncube) [inline libs/math/src/math/linalg/blas/gemm_tn_novector.cpp]: "Type\<Type\>" looks really tough ))

File: libs/math/src/math/linalg/blas/gemm_tn_vector.cpp [hunk 2]
```diff
@@ -35,19 +35,18 @@ void Blas<S, Signature(_C <= _alpha, _A, _B, _beta, _C),
...
-  if (alpha == static_cast<Type>(0.0))
+  if (alpha == Type{0})
   {
-    if (beta == static_cast<Type>(0.0))
+    if (beta == Type{0})
     {
       for (j = 0; j < c.width(); ++j)
       {
-        Type zero{0.0};
+        auto zero = Type{0};
```
Comment (vhavrylov-ncube) [inline libs/math/src/math/linalg/blas/gemm_tn_vector.cpp:49]: I would suggest "auto const", if  zero is not changed later

**Review by vhavrylov-ncube — DISMISSED** [review]
LGTM

File: libs/math/src/math/linalg/blas/gemm_tn_novector.cpp [hunk 3]
```diff
@@ -38,21 +38,21 @@ void Blas<S, Signature(_C <= _alpha, _A, _B, _beta, _C),
...
   {
-    if (beta == static_cast<Type>(0.0))
+    if (beta == fetch::math::Type<Type>("0.0"))
     {
       for (j = 0; j < c.width(); ++j)
       {
         for (i = 0; i < c.height(); ++i)
         {
-          c(i, j) = static_cast<Type>(0.0);
+          c(i, j) = fetch::math::Type<Type>("0.0");
```
Comment (vhavrylov-ncube) [inline libs/math/src/math/linalg/blas/gemm_tn_novector.cpp]: blas is a performance critical part, it should not have any string parser invoked anywhere

**Review by markos — APPROVED** [review]
Looks good

**Review by baykaner — APPROVED** [review]

**Pull request merged**